### DISCUSSION TITLE
Offline maps with hillshading

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#558] [Q&A] [answered]
[Cruiser 5](https://github.com/devemux86/cruiser/discussions/559) will have offline maps with hillshading and digital elevation model data.

- Available in "Settings | Map | Hillshading"
where hillshading magnitude can also be adjusted

- Hillshading data is optionally downloaded with all offline maps

- Or use the menu "Download hillshading data" in the local maps

More information:
- https://viewfinderpanoramas.org/dem3.html
- https://sonny.4lima.de

<img src="https://github.com/user-attachments/assets/347df58e-983e-46ef-8fd7-a07222b36641" width="50%"/><img src="https://github.com/user-attachments/assets/7110f3ac-5c3a-48c3-8df3-343b3bebc840" width="50%"/>

**Comment by devemux86** [comment]
I updated the topic with more information about the new offline hillshading and some images.

**Reply by kds112** [reply]
Will the new hillshading work with OAM maps (and Elevate theme) as well or with Mapsforge maps only?

**Reply by DIrgi** [reply]
Will it be possible to define the intensity grade of the hill shading?

**Reply by devemux86** [reply]
@kds112 

> Will the new hillshading work with OAM maps (and Elevate theme) as well or with Mapsforge maps only?

The new hillshading can be used with all offline maps, I updated the description [above](https://github.com/devemux86/cruiser/discussions/558#discussion-6747347).

Hillshading (and BRouter) data can be downloaded automatically with the Mapsforge and Freizeitkarte maps,
because Mapsforge and Freizeitkarte provide the map bounds, so hillshading (and BRouter) data can be found.

I have to see what is possible with OpenAndroMaps.

We can always check and download manually the hillshading (and BRouter) data, with below menu:

<img src="https://github.com/devemux86/cruiser/assets/3484020/02b94afd-2fcb-4fb8-86ec-5fc95bba96eb" width="50%"/>

**Reply by devemux86** [reply]
@DIrgi 

> Will it be possible to define the intensity grade of the hill shading?

Probably the hillshading opacity, the new map engine is very powerful. :slightly_smiling_face:

**Comment by devemux86** [comment]
<img src="https://github.com/devemux86/cruiser/assets/3484020/cf5510b9-1fe3-4570-a74c-ef663d8f001f" width="50%"/>

**Comment by devemux86** [comment]
Update:

From now on also with the **Freizeitkarte** maps we will have automatic downloads of hillshading and BRouter data.

I contacted OpenAndroMaps to see what can be automated with their maps.

**Reply by DolomitesTom** [reply]
Sounds good, where can this version be downloaded?

**Reply by devemux86** [reply]
@DolomitesTom very soon! :slightly_smiling_face:

**Reply by kds112** [reply]
It looks like that OAM will support your request 👍 (see OAM forum reply)

**Reply by devemux86** [reply]
@kds112 I have seen it, this is very nice! :slightly_smiling_face:

**Comment by devemux86** [comment]
Update:

From now on also with the **OpenAndroMaps** maps we will have automatic downloads of hillshading and BRouter data.
(POI are also included)

Thanks to Christian and the other OpenAndroMaps team for their quick response to my request. :+1:

**Comment by devemux86** [comment]
Offline maps with hillshading are now available in:

- https://github.com/devemux86/cruiser/discussions/610

- Instructions: https://github.com/devemux86/cruiser/discussions/558#discussion-6747347

**Comment by devemux86** [comment]
Mapsforge community contributed a new hillshading algorithm
without gray shades where there are no slopes.

It will be included in the next Cruiser update.

You can see the images (old vs new) here:
- https://github.com/mapsforge/mapsforge/discussions/1512#discussioncomment-10347296

**Reply by bios4** [reply]
Did this new algorithm make the [hill shading opacity setting](https://github.com/devemux86/cruiser/discussions/558#discussioncomment-9912289) obsolete?
It is gone in Cruiser 5.0.0-b8...

**Reply by devemux86** [reply]
It works differently, don't know if we even need such setting and how much it can change the map.

**Reply by bios4** [reply]
So it's gone on purpose, that's what I wanted to know... 😀

**Comment by devemux86** [comment]
Both versions Cruiser b7 + b8 are available.

You can compare the hillshading and tell me how each version looks to you.
(it's easier on desktop)

**Comment by devemux86** [comment]
Hillshading magnitude setting will be available again in the next update with interesting visual results:

<img src="https://github.com/user-attachments/assets/347df58e-983e-46ef-8fd7-a07222b36641" width="50%"/><img src="https://github.com/user-attachments/assets/7110f3ac-5c3a-48c3-8df3-343b3bebc840" width="50%"/>

**Comment by devemux86** [comment, marked as answer]
- Offline maps with hillshading are available in https://github.com/devemux86/cruiser/discussions/641

For instructions please see [above](https://github.com/devemux86/cruiser/discussions/558#discussion-6747347).

**Comment by umpfmpf** [comment]
Is it possible to display hillshading even with a lower zoom? Maybe as part of the Woldmap!?

**Reply by devemux86** [reply]
It could be possible, however it will need to download too many hillshading data files.
Hillshading is more useful when also seeing other natural data, like forests, lakes, etc.
These are available in the detailed offline maps of the countries.

**Comment by devemux86** [comment]
We can download routing and hillshading data on the screen of downloaded offline local maps.
It's convenient as it checks the bounds of the offline map, like when we first download the map.

It will be available in the next update.

<img src="https://github.com/user-attachments/assets/56d8defe-825c-47b0-b099-390dd6e5b2bd" width="50%"/>

**Reply by kds112** [reply]
Please release the beta to test it 👍🎉🏁 Good simplification and intuitive!

**Reply by devemux86** [reply]
It will be available in the next Beta update.

**Reply by devemux86** [reply]
> We can download routing and hillshading data on the screen of downloaded offline local maps.

Available in **Cruiser 5.0.1-b15**.

**Reply by kds112** [reply]
It works👍 Thanks a lot!

**Comment by devemux86** [comment]
> We can download routing and hillshading data on the screen of downloaded offline local maps.

- Available in https://github.com/devemux86/cruiser/discussions/641

**Comment by MotoUKRider** [comment]
Is it possible to parse the data in the HGT hillshading files to make the Cruiser map "elevation aware"?
So any touch on the map produces not only the Lat/Lon, but also the Elevation for use as a "how high is that road" or when placing a Favorite.

I find this information very useful, and have been a little spoiled with BaseCamp offering real-time Lat/Lon+Elevation as I move my mouse.

**Reply by devemux86** [reply]
@MotoUKRider It is possible if we implement it first in Mapsforge: :slightly_smiling_face: 
- https://github.com/mapsforge/mapsforge/issues/1621

It is also possible to request BRouter (offline) data for elevation information.
But as BRouter stores the road network, it may not be available everywhere.

**Reply by devemux86** [reply]
@MotoUKRider I posted your suggestion in a new thread to discuss it there:
- https://github.com/devemux86/cruiser/discussions/703

**Comment by bios4** [comment]
I have Cruiser installed on three devices.
On all of them I'm using the VTM map engine and the "Mapy.cz" map theme from JimmyS83.

But on the one device I use on the motorcycle (Samsung Galaxy J6, Android 10), the hillshading display doesn't work - neither with Cruiser 5.3.0 (was installed until today morning), nor with Cruiser 5.4.0 (installed today).

Cruiser 5.4.1 isn't available for me on Google Play yet.

So my question @devemux86 is: is the hillshading functionality tied to a specific (minimal) Android version, or are there any other restrictions I am not aware of?

**Reply by devemux86** [reply]
Hillshading works on all Android versions.

@bios4 
Have you downloaded the hillshading data for the map?

You can try to force the hillshading to be created again:
- Clear the app cache in Android settings
- Or change the hillshading magnitude in app settings

**Reply by devemux86** [reply]
@bios4 

Or delete the hgt folder and download the hillshading data again.

**Reply by bios4** [reply]
> Have you downloaded the hillshading data for the map?

Yes of course; also tried again before writing my post and it said "Data already present" (as expected).

> Clear the app cache in Android settings
> Or change the hillshading magnitude in app settings

Tried also this before, with no effect.

> Or delete the hgt folder and download the hillshading data again.

Followed this ^^ advice now, and now the hillshading works also on this device.
I have no idea what caused it to fail before...

Thanks.

**Reply by devemux86** [reply]
@bios4 Probably the hillshading data were corrupted or downloaded / written wrong.

It would be interesting to check the files that are not working, if you have kept them.

**Reply by bios4** [reply]
> It would be interesting to check the files that are not working, if you have kept them.

Unfortunately I wiped the data completely before re-downloading them.
Did not think that they might be of any use... sorry!

**Reply by devemux86** [reply]
"Download hillshading" checks the existence of the required files for the region.
It does not open every file or check if it is "bad" in one way or another.
